frontend/src/plugins/router/index.ts

```diff
@@ -30,6 +31,19 @@ router.beforeEach(metaGuard);
 router.beforeEach(loginGuard);
 router.beforeEach(adminGuard);
 router.beforeEach(validateGuard);
+router.beforeEach(playbackGuard);
+
+/**
+ * Replaces the 'back' function, taking into account if there's a previous page or not.
+ * If there's no previous page in history, we ensure we want to go home
+ */
+router.back = (): ReturnType<typeof router.back> => {
+  router.replace(
+    typeof router.options.history.state.back === 'string'
+      ? router.options.history.state.back
+      : '/'
```

Comment:
For my own curiosity, what was the reason this had to be done?

Reply:
@aweebs Try this:

* Go to incognito window (just to be sure we don't mess with cache)
* Access ``google.com``
* Open a new tab and login at ``vite.jf-vue.pages.dev``. Close that tab after you're in the home screen
* Go back to the tab with Google and **replace** the url with ``vite.jf-vue.pages.dev/#/playback/music``
* As no playback will be happening, you'll be redirected back, going to Google instead of the bounds of the client

Repeat the same steps but with the CloudFlare deployment URL of this PR. You'll be redirected back to th home page of the client instead. This is what replacing the function does, strictly staring within the bounds of the client.